### PR TITLE
ADD: Testcase to show broken java.sql.Time support

### PR DESCRIPTION
**Pull request by rPraml**
This PR is for issue #559

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+import com.microsoft.sqlserver.jdbc.SQLServerPreparedStatement;
+import com.microsoft.sqlserver.jdbc.SQLServerStatement;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.DBConnection;
+import com.microsoft.sqlserver.testframework.Utils;
+
+@RunWith(JUnitPlatform.class)
+public class DateAndTimeTypeTest extends AbstractTest {
+
+    private static final Date DATE_TO_TEST = new java.sql.Date(2017, 20, 11);
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java]: Could we use the non-deprecated constructors for Date, Time and Timestamp?

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+    }
+    private void createTVPs(String tvpName, String tvpType) throws SQLException {
+      stmt.executeUpdate("IF EXISTS (SELECT * FROM sys.types WHERE is_table_type = 1 AND name = '" + tvpName + "') " + " drop type " + tvpName);
+      String TVPCreateCmd = "CREATE TYPE " + tvpName + " as table (c1 " + tvpType + " null)";
+      stmt.executeUpdate(TVPCreateCmd);
+    }
+
+    @BeforeEach
+    public void testSetup() throws TestAbortedException, Exception {
+        assumeTrue(9 <= new DBConnection(connectionString).getServerVersion(),
```
Comment (peterbae) [inline src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java]: I think we can close this DBConnection by using try statement.

Comment (rPraml) [reply src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java]: Ack. (I think I've copied this line from an other test)

**Comment by rPraml** [comment]
Hello @peterbae can you update this PR as mentioned above or should I do that?

File: src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java (hunk 1 above)
Comment (peterbae) [reply src/test/java/com/microsoft/sqlserver/jdbc/datatypes/DateAndTimeTypeTest.java]: I think this style of argument is deprecated, and they want us to provide a long value that corresponds to this time instead. The long values for these Date, Time and Timestamp objects come to 61494793200000L, 74096000L, and 61494838496000L respectively, I believe.

**Comment by peterbae** [comment]
@rPraml I can update this PR for you, and I'll merge it as well.

**Pull request merged**